proposals/0219-ios-check-type.md

```diff
@@ -30,18 +30,33 @@ For example:
 
 ## Proposed solution
 
-Remove method :
+Mark method as private:
 `-(nullable id)sdl_objectForName:(SDLName)name;`. 
 
-Instead of the removed method, use this existing method:
+Instead of method above use next methods:
+
+For optional values, can be used functions that return object of classType or nil:
 `- (nullable id)sdl_objectForName:(SDLName)name ofClass:(Class)classType;`.
+`- (nullable NSArray *)sdl_objectsForName:(SDLName)name ofClass:(Class)classType`.
+
+For mandatory values, next functions try to return object of classType overwise error would be filled. 
+In case, when error isn't `nil`, returned object can be not type of `classType`:
+`- (nullable id)sdl_objectForName:(SDLName)name ofClass:(Class)classType error:(NSError **)error`
+`- (nullable NSArray *)sdl_objectsForName:(SDLName)name ofClass:(Class)classType error:(NSError **)error;`.
 
 For suitable working with SDLEnums create the following methods:
-`- (nullable SDLEnum)sdl_enumForName:(SDLName)name`
+`- (nullable SDLEnum)sdl_enumForName:(SDLName)name;`
```

Comment:
Is this needed? it's not included in the requested revisions.

Reply:
In my view, we can save this method if we need to getting SDLEnum and don't want to handle error.
```
- (nullable SDLEnum)sdl_enumForName:(SDLName)name {
    return [self sdl_objectForName:name ofClass:NSString.class];
}
- (nullable id)sdl_objectForName:(SDLName)name ofClass:(Class)classType {
    NSError *error;
    id object = [self sdl_objectForName:name ofClass:classType error:&error];
    return error == nil ? object : nil;
}
```